tools/generate-bedrock-data-from-packets.php

```diff
@@ -234,31 +253,75 @@ private static function sort(mixed $object) : mixed{
 	}
 
 	public function handleStartGame(StartGamePacket $packet) : bool{
-		$this->itemTypeDictionary = new ItemTypeDictionary($packet->itemTable);
+		foreach(Utils::promoteKeys($packet->levelSettings->experiments->getExperiments()) as $name => $experiment){
+			echo "Experiment \"$name\" is " . ($experiment ? "" : "not ") . "active\n";
+		}
+		return true;
+	}
+
+	public function handleItemRegistry(ItemRegistryPacket $packet) : bool{
+		$this->itemTypeDictionary = new ItemTypeDictionary($packet->getEntries());
 
 		echo "updating legacy item ID mapping table\n";
+		$emptyNBT = new CompoundTag();
 		$table = [];
-		foreach($packet->itemTable as $entry){
+		foreach($packet->getEntries() as $entry){
 			$table[$entry->getStringId()] = [
 				"runtime_id" => $entry->getNumericId(),
-				"component_based" => $entry->isComponentBased()
+				"component_based" => $entry->isComponentBased(),
+				"version" => $entry->getVersion(),
 			];
+
+			$componentNBT = $entry->getComponentNbt()->getRoot();
+			if(!$componentNBT->equals($emptyNBT)){
+				$table[$entry->getStringId()]["component_nbt"] = base64_encode((new LittleEndianNbtSerializer())->write(new TreeRoot($componentNBT)));
+			}
 		}
 		ksort($table, SORT_STRING);
 		file_put_contents($this->bedrockDataPath . '/required_item_list.json', json_encode($table, JSON_PRETTY_PRINT) . "\n");
 
-		foreach(Utils::promoteKeys($packet->levelSettings->experiments->getExperiments()) as $name => $experiment){
-			echo "Experiment \"$name\" is " . ($experiment ? "" : "not ") . "active\n";
-		}
+		echo "updating item registry\n";
+		$items = array_map(function(ItemTypeEntry $entry) : array{
+			return self::objectToOrderedArray($entry);
+		}, $packet->getEntries());
+		file_put_contents($this->bedrockDataPath . '/item_registry.json', json_encode($items, JSON_PRETTY_PRINT) . "\n");
 		return true;
 	}
 
 	public function handleCreativeContent(CreativeContentPacket $packet) : bool{
 		echo "updating creative inventory data\n";
-		$items = array_map(function(CreativeContentEntry $entry) : array{
-			return self::objectToOrderedArray($this->itemStackToJson($entry->getItem()));
-		}, $packet->getEntries());
-		file_put_contents($this->bedrockDataPath . '/creativeitems.json', json_encode($items, JSON_PRETTY_PRINT) . "\n");
+
+		/** @var array<int, ItemStackData[]> $groupItems */
```

Comment:
```suggestion
		/**
		 * @var ItemStackData[][] $groupItems
		 * @phpstan-var array<int, list<ItemStackData>> $groupItems
		 */
```

We don't use generics in general @var, they aren't part of the PSR standard